core/node/node_framework/src/wiring_layer.rs

```diff
@@ -0,0 +1,26 @@
+use crate::{node::NodeContext, resource::ResourceId};
+
+/// Wiring layer provides a way to customize the `ZkStackService` by
+/// adding new tasks or resources to it.
+///
+/// Structures that implement this trait are advised to specify in doc comments
+/// which resources they use or add, and the list of tasks they add.
```

Comment:
Kinda. If something requires a non-lazy resource and it's not provided -> it will fail at runtime.
If something requires a lazy resource and this resource is not provided -> resource won't be resolved.

However, the docs are read before you reach the runtime, so reading the docs is how you know how you should compose the node so that it doesn't fail at runtime.